packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -333,9 +333,10 @@ class TestCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts {
       testTimeRecorder = TestTimeRecorder(globals.logger);
     }
 
-    if (buildInfo.packageConfig['test_api'] == null) {
+    if (!flutterProject.manifest.devDependencies.contains('flutter_test') &&
```

Comment:
should we not rely on the packageConfig, rather than the pubspec.yaml (I'm assuming `flutterProject.manifest.devDependencies` is the parsed pubspec.yaml)?

Reply:
(this was a drive-by question btw, I didn't actually dig into the code here)

Reply:
Ah yep that works as well, I didn't realize that an entry was getting generated for `flutter_test` in the `package_config.json`